doozerlib/runtime.py

```diff
@@ -1106,6 +1130,8 @@ def resolve_source(self, meta):
                         self.logger.info(f"Fetching commit-ish {meta.commitish} from upstream")
                         cmd = ["git", "-C", source_dir, "fetch", "origin", "--depth", "1", meta.commitish]
                         exectools.cmd_assert(cmd)
+                    exectools.cmd_assert(["git", "-C", source_dir, "checkout", meta.commitish])
```

Comment:
@vfreex can you take a look at this? I was exercising `--upstream`, but found that it didn't leave the cloned source checked out at the expected commit.  Does this look appropriate to you?